ask/commands.py

```diff
@@ -0,0 +1,182 @@
+import sublime
+import sublime_plugin
+import pieces_os_client as pos_client
+from pieces.settings import PiecesSettings
+import re
+from difflib import Differ 
+from .prompts import *
+import mdpopups
+import time
+LIST_ITEMS = [
+	("Fix a bug",BUGS_PROMPT),
+	("Find a possible bug", POSSIBLE_BUGS_PROMPT),
+	("Write cleaner code", CLEANER_CODE_PROMPT),
+	("Add a doc string to this function",DOC_STRING_PROMPT),
+	("Add comments to the code",ADD_COMMENTS_PROMPT)
+] # TODO: Add more suff 
+
+
+class DescriptionInputHandler(sublime_plugin.TextInputHandler):
+	def placeholder(self):
+		return "Enter a quick description of the bug itself or useful error message"
+	def validate(self, text):
+		return len(text) > 0
+
+class QuestionInputHandler(sublime_plugin.ListInputHandler):
+	def list_items(self):
+		return LIST_ITEMS
+	def next_input(self,args):
+		if args in description_needed_commands:
+			return DescriptionInputHandler()
+
+
+
+
+class PiecesAskQuestionCommand(sublime_plugin.TextCommand):
+
+	def run(self,edit, question, description=None):
+		sublime.set_timeout_async(lambda:self.run_async(edit,question,description),0)
+
+		self.view.set_status('Pieces Refactoring', 'Copilot is thinking...')
+		
+
+	def run_async(self,edit, question, description):
+		# Get the current selection
+		self.selection = self.view.sel()[0]
+		selected_text = self.view.substr(self.selection)
+		
+		if "." in self.view.file_name():
+			self.langauge = self.view.file_name().split(".")[-1]
+		else: 
+			self.langauge = "txt"
+
+		if not selected_text:
+			return # TODO show error
+
+		if question in description_needed_commands:
+			query = question.format(description=description,code=selected_text)
+		else:
+			query = question.format(code=selected_text)
+		self.view.set_status('Pieces Refactoring', 'Copilot analyzing...')
+		res = pos_client.QGPTApi(PiecesSettings.api_client).question(
+			pos_client.QGPTQuestionInput(
+				query = query,
+				model = PiecesSettings.model_id,
+				relevant = pos_client.RelevantQGPTSeeds(
+					iterable = [
+					# TODO: Add the snippet as a context
```

Comment:
nice it's a good rule of thumb if we have commented out code in the project to include a note about why the code should stay